webapp/src/app.tsx

```diff
@@ -1148,7 +1148,8 @@ export class ProjectView
                         if (e.notifyUser) {
                             core.warningNotification(e.message);
                         } else {
-                            core.warningNotification(lf("Upload failed, please try again."));
+                            const errorText = pxt.appTarget.appTheme.useUploadMessage ? lf("Upload failed, please try again.") : lf("Download failed, please try again.");
```

Comment:
Can appTheme be null?

Reply:
It can be null in the CLI but not in the app itself.